drivers/audio/wm8994.c

```diff
@@ -1374,24 +1890,324 @@ static void *wm8994_workerthread(pthread_addr_t pvarg)
+               WM8994_WSEQ_START |
+               (0x0 << WM8994_WSEQ_START_INDEX_SHIFT);  /* Start Index = 0 */
+      wm8994_writereg(priv, WM8994_WR_CTL_SEQ1, regval);
+      up_mdelay(20);
+
+      /* Wait until sequencer indicates that sequence is completed */
+
+      regval = wm8994_readreg(priv, WM8994_WR_CTL_SEQ2);
+      while (regval & WM8994_WSEQ_BUSY)
+        {
+          regval = wm8994_readreg(priv, WM8994_WR_CTL_SEQ2);
+          up_mdelay(20);
+        }
+
+      /* TODO: Manage cold/warm start correctly */
+
+      cold_startup = 0;
+    }
+  else /* Headphone Warm Start-Up */
+    {
+      regval = WM8994_WSEQ_ENA |
+               WM8994_WSEQ_START |
+               (0x8 << WM8994_WSEQ_START_INDEX_SHIFT);  /* Start Index = 8 */
+      wm8994_writereg(priv, WM8994_WR_CTL_SEQ1, regval);
+      up_mdelay(20);
+
+      /* Wait until sequencer indicates that sequence is completed */
+
+      regval = wm8994_readreg(priv, WM8994_WR_CTL_SEQ2);
+      while (regval & WM8994_WSEQ_BUSY)
+        {
+          regval = wm8994_readreg(priv, WM8994_WR_CTL_SEQ2);
+          up_mdelay(20);
+        }
+    }
+
+  /* Soft un-Mute the AIF1 Timeslot 0 DAC1 path L&R */
+
+  regval = 0x0000;
+  wm8994_writereg(priv, WM8994_AIF1_DAC1_FILTERS1, regval);
+
+  /* Enable SPKRVOL PGA, Enable SPKMIXR, Enable SPKLVOL PGA, Enable SPKMIXL */
+
+  regval = WM8994_SPKRVOL_ENA |
+    WM8994_SPKLVOL_ENA;
+  wm8994_writereg(priv, WM8994_PM3, regval);
+
+  /* Left Speaker Mixer Volume = 0dB */
+
+  regval = 0x0000;
+  wm8994_writereg(priv, WM8994_SPKMIXL_ATT, regval);
+
+  /* Speaker output mode = Class D,
+   * Right Speaker Mixer Volume = 0dB
+   */
+
+  regval = 0x0000;
+  wm8994_writereg(priv, WM8994_SPKMIXR_ATT, regval);
+
+  if (spk_out)
+    {
+      /* Unmute DAC2 (Left) to Left Speaker Mixer (SPKMIXL) path,
+       * Unmute DAC2 (Right) to Right Speaker Mixer (SPKMIXR) path
+       */
+
+      regval = WM8994_DAC2L_TO_SPKMIXL |
+               WM8994_DAC2R_TO_SPKMIXR;
+      wm8994_writereg(priv, WM8994_SPEAKER_MIXER, regval);
+
+      /* Enable bias generator, Enable VMID, Enable SPKOUTL, Enable SPKOUTR */
+
+      regval = WM8994_SPKOUTR_ENA |
+               WM8994_SPKOUTL_ENA |
+               WM8994_BIAS_ENA |
+               WM8994_VMID_SEL_2X40K;
+      wm8994_writereg(priv, WM8994_PM1, regval);
+
+      /* Enable Class W, Class W Envelope Tracking = AIF1 Timeslot 0 */
+
+      regval = WM8994_CP_DYN_PWR;
+      regval = 0x0005;  /* TODO: Check where this comes from? */
+      wm8994_writereg(priv, WM8994_CLASS_W_1, regval);
+    }
+
+  /* Enable normal bias generator, Enable VMID */
+
+  regval =  WM8994_BIAS_ENA |
+            WM8994_VMID_SEL_2X40K;
+
+  /* Enable speaker */
+
+  if (spk_out)
+    regval |= WM8994_SPKOUTR_ENA |
+              WM8994_SPKOUTL_ENA;
+
+  /* Enable HPOUT1 (Left) and
+   * Enable HPOUT1 (Right) input stages
+   */
+
+  if (hp_out)
+    regval |= WM8994_HPOUT1L_ENA |
+              WM8994_HPOUT1R_ENA;
```

Comment:
please wrap with `{}`

Reply:
could you please address this coment?

Reply:
Sorry, missed that one. Hope to have that resolved by now. Thanks for the review and the explanation regarding the use of FAR!